File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.js
function solution(s) {
    var answer = 0;
     while (s.length > 0) {
        first=s[0];
        firstCount=0;
        otherCount=0;
        
        for (let i = 0; i < s.length; i++) {
            if (s[i]==first){
                firstCount++;
            }else{
                otherCount++;
            }
            if(firstCount==otherCount){
                s=s.substring(i+1)
                answer++;
                break;
            }
            if (i === s.length - 1) {
                s = "";  
                answer++;
            }
        }
    }
    return answer;
}

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    answer = 0
    i = 0
    while i < len(s):
        first = s[i]
        firstCount = 0
        otherCount = 0
        for j in range(i, len(s)):
            if s[j] == first:
                firstCount += 1
            else:
                otherCount += 1
            
            if firstCount == otherCount:
                answer += 1
                i = j + 1  
                break
        else:
            answer += 1
            break 
    
    return answer